Complete task02

GushchinGleb opened the pull request:
Complete task 2.

nevidimka787 committed:
Complete task02

shadowk1337 commented on gushchin_g/task02/src/task02/task02.py:
```diff
@@ -0,0 +1,115 @@
+import sys
+from typing import List
+from os import walk
```
Обычно импортуруют весь модуль `os`

shadowk1337 commented on gushchin_g/task02/src/task02/task02.py:
```diff
@@ -0,0 +1,115 @@
+from os import walk
+
+PATH_ARGUMNET_ID = 2
+REQURSION_ARGUMENT_ID = 1
```
`RECURSION`

shadowk1337 commented on gushchin_g/task02/src/task02/task02.py:
```diff
@@ -0,0 +1,115 @@
+PATH_ARGUMNET_ID = 2
+REQURSION_ARGUMENT_ID = 1
+
+STYLE_RESSET = "\033[0m"
```
`RESET`

shadowk1337 commented on gushchin_g/task02/src/task02/task02.py:
```diff
@@ -0,0 +1,115 @@
+        if element is False:
+            append = INTEDATION_EMPTY
+        else:
+            append = INTEDATION_VERTICAL
```
Можно упростить с использованием тернарного оператора

shadowk1337 commented on gushchin_g/task02/src/task02/task02.py:
```diff
@@ -0,0 +1,115 @@
+STYLE_RESSET = "\033[0m"
+STYLE_COLOR_BLUE = "\033[34m"
+
+INTEDATION_EMPTY = "    "
```
`INDENTATION`

shadowk1337 commented on gushchin_g/task02/src/task02/task02.py:
```diff
@@ -0,0 +1,115 @@
+
+
+def generate_intedation(lines: List[bool]) -> str:
+    answer: str = ""
```
не очень удачное название переменной, наверное лучше назвать ее `result`

shadowk1337 commented on gushchin_g/task02/src/task02/task02.py:
```diff
@@ -0,0 +1,115 @@
+INTEDATION_END = "|-- "
+
+PATH_ID = 0
+TYPE_ID = 1
```
Это лучше вынести в енам

shadowk1337 commented on gushchin_g/task02/src/task02/task02.py:
```diff
@@ -0,0 +1,115 @@
+            if dir_iter >= dirs_count:
+                lines.append(False)
+            else:
+                lines.append(True)
```
```python
lines.append(dir_iter >= dirs_count)
```

shadowk1337 commented on gushchin_g/task02/src/task02/task02.py:
```diff
@@ -0,0 +1,115 @@
+        return ""
+
+
+def get_reqursion_number(arguments: List[str]) -> int:
+    try:
+        answer = int(arguments[REQURSION_ARGUMENT_ID])
+        if answer < 0:
+            return 0
+        return answer
+    except Exception:
+        return 0
```
_Optional_
Такие параметры лучше передавать через флаги

shadowk1337 commented on gushchin_g/task02/src/task02/task02.py:
```diff
@@ -0,0 +1,115 @@
+    dir_iter: int = 0
+    for element in union_list:
+        name: str = element[PATH_ID]
+        if element[TYPE_ID] is True:
```
Тут уже не понятно и забывается, кто такой True - директория или файл, один из вариантов вынести это в константу

shadowk1337 commented on gushchin_g/task02/src/task02/task02.py:
```diff
@@ -0,0 +1,115 @@
+        if element[TYPE_ID] is True:
+            name = set_style(element[PATH_ID], STYLE_COLOR_BLUE)
+            dir_iter += 1
+        print(f"{indentation}{name}")
```
Вывод лучше вынести из этой функции и сделать, как отдельную функцию, разве имя `scan_dirrectory` подразумевает, что эта функция что-то выводит в консоль?

shadowk1337 commented on gushchin_g/task02/src/task02/task02.py:
```diff
@@ -0,0 +1,115 @@
+    check_path: str = get_path(sys.argv)
+    reqursion: int = get_reqursion_number(sys.argv)
+
+    if reqursion <= 0:
+        exit()
```
Эта проверка у тебя происходит в `get_reqursion_number` зачем еще раз  проверять

shadowk1337 reviewed (changes requested):
Надо поправить недочеты и поправить названия переменных (не только, которых я указал)

nevidimka787 committed:
Fix task 2

Add enumerators insted of constants.
Add green color to files' names.
Change some variables' names.
Update tree style.

shadowk1337 reviewed (approved):
Sum: 3/3
